src/main/scala/is/hail/methods/LDPrune.scala

```diff
@@ -403,13 +483,28 @@ object LDPrune {
     })
     info(s"LD prune step 2 of 3: nVariantsKept=$nVariantsLP2, nPartitions=$nPartitionsLP2, time=${ formatTime(durationLP2) }")
 
-
     val ((globalPrunedRDD, nVariantsFinal), globalDuration) = time(pruneGlobal(rddLP2, r2Threshold, windowSize))
     info(s"LD prune step 3 of 3: nVariantsKept=$nVariantsFinal, time=${ formatTime(globalDuration) }")
 
-    vds.copyLegacy(
-      rdd = vds.typedRDD[Locus, Variant, Annotation]
-        .orderedInnerJoinDistinct(globalPrunedRDD)
-        .mapValues { case ((va, gs), _) => (va, gs) })
+    vsm.copy2(rdd2 = OrderedRDD2(vsm.rdd2.typ, vsm.rdd2.orderedPartitioner, vsm.rdd2.orderedJoinDistinct(globalPrunedRDD, "inner").mapPartitions({ it =>
```

Comment:
I kind of feel like `ld_prune` should just return a list of pruned variants and the user can do the inner join if they want.

Reply:
I'd like to save this for a future PR.

Reply:
OK.